packages/engine/lib/memory/src/shared_memory/segment/cleanup.rs

```diff
@@ -1,48 +1,111 @@
+use std::{collections::HashSet, sync::Mutex};
+
 use glob::GlobError;
+use lazy_static::lazy_static;
 use uuid::Uuid;
 
-use super::MemoryId;
-use crate::{Error, Result};
+lazy_static! {
+    /// We use this to keep a list of all the shared-memory segements which are
+    /// being used by the engine. When the engine exits, we then delete any
+    /// leftover shared memory segments (in release builds; we error in debug builds).
+    pub static ref IN_USE_SHM_SEGMENTS: Mutex<HashSet<String>> = Mutex::new(HashSet::new());
```

Comment:
`Mutex::new` is const since Rust 1.63, so probably no need for `lazy_static` at all.

Reply:
`HashSet` does not have `const new`, but [`BTreeSet`](https://doc.rust-lang.org/nightly/alloc/collections/btree_set/struct.BTreeSet.html) has 🙂

Reply:
But because it is constant, wouldn't you be able to use: `static IN_USE_SHM_SEGMENTS: Mutex<HashSet<String>> = Mutex::new(HashSet::new())` instead (not using `lazy_static`)?

`const` functions can also be invoked at a `static` level without a problem.